lib/puppet/util/windows/taskscheduler2_v1task.rb

```diff
@@ -0,0 +1,559 @@
+  #
+  # Note that if intending to use SYSTEM, specify an empty user and nil password
+  #
+  # Calling task.set_account_information('SYSTEM', nil) will generally not
+  # work, except for one special case where flags are also set like:
+  # task.flags = Win32::TaskScheduler::TASK_FLAG_RUN_ONLY_IF_LOGGED_ON
+  #
+  # This must be done prior to the 1st save() call for the task to be
+  # properly registered and visible through the MMC snap-in / schtasks.exe
+  #
+  def set_account_information(user, password)
+    @task.set_principal(user, password)
+  end
+
+  def account_information
+    principal = @task.principal
+
+    principal.nil? ? nil : principal.UserId
+  end
+
+  def application_name
+    action = default_action
+    action.nil? ? nil : action.Path
+  end
+
+  def application_name=(app)
+    action = default_action(true)
+    action.Path = app
+
+    app
+  end
+
+  def parameters
+    action = default_action
+    action.nil? ? nil : action.Arguments
+  end
+
+  def parameters=(param)
+    action = default_action(true)
+    action.Arguments = param
+
+    param
+  end
+
+  def working_directory
+    action = default_action
+    action.nil? ? nil : action.WorkingDirectory
+  end
+
+  def working_directory=(dir)
+    action = default_action
+    action.WorkingDirectory = dir
+
+    dir
+  end
+
+  def priority
+    @task.priority
+  end
+
+  def priority=(value)
+    raise TypeError unless value.is_a?(Numeric)
+
+    @task.priority_value = value
+
+    value
+  end
+
+  # Creates a new work item (scheduled job) with the given +trigger+. The
+  # trigger variable is a hash of options that define when the scheduled
+  # job should run.
+  #
+  def new_work_item(task_name, task_trigger)
+    raise TypeError unless task_trigger.is_a?(Hash)
+
+    task_name = Puppet::Util::Windows::TaskScheduler2::ROOT_FOLDER + validate_task_name(task_name)
+
+    @task.new_task_defintion(task_name)
+
+    @task.compatibility = Puppet::Util::Windows::TaskScheduler2::TASK_COMPATIBILITY_V1
+
+    append_trigger(task_trigger)
+
+    set_account_information('',nil)
+
+    @task.definition
+  end
+
+  alias :new_task :new_work_item
+
+  def trigger_count
+    @task.trigger_count
+  end
+
+  def delete_trigger(v1index)
+    # The older V1 API uses a starting index of zero, wherease the V2 API uses one.
+    # Need to increment by one to maintain the same behavior
+    @task.delete_trigger(v1index + 1)
+  end
+
+  # TODO Need to convert the API v2 style triggers into API V1 equivalent hash
+  def trigger(v1index)
+    # The older V1 API uses a starting index of zero, wherease the V2 API uses one.
+    # Need to increment by one to maintain the same behavior
+    populate_v1trigger(@task.trigger(v1index + 1))
+  end
+
+  # Sets the trigger for the currently active task.
+  #
+  # Note - This method name is a mis-nomer. It's actually appending a newly created trigger to the trigger collection.
+  def trigger=(v1trigger)
+    append_trigger(v1trigger)
```

Comment:
Fortunately, this method is much better at describing what is actually happening.

However, the original API of `.trigger = ` as something that adds new triggers to the end of a running list of triggers is totally bonkers 👎 .  We should fix both of those, but we can ticket that / do it later.

https://github.com/puppetlabs/puppet/blob/master/lib/puppet/provider/scheduled_task/win32_taskscheduler.rb#L192-L196